File: frontend/store/types/discount.ts
export enum DiscountType {
  PERCENTAGE = 'percentage',
  FIXED = 'fixed',
}

export enum DiscountApplyType {
  PRODUCT = 'product',
  ORDER = 'order',
  SHIPPING = 'shipping',
}

export enum DiscountStatus {
  ACTIVE = 'active',
  SCHEDULED = 'scheduled',
  PAUSED = 'paused',
  ENDED = 'ended',
}

export type Discount = {
  id: string;
  name: string;
  code: string;
  type: DiscountType;
  value: number;
  applyTo: DiscountApplyType;
  minOrder: number;
  usageLimit: number;
  usagePerUser: number;
  usageCount: number;
  startTime: Date;
  endTime: Date;
};

File: backend/app/core/enums/discount.py
from enum import Enum


class DiscountType(str, Enum):
    PERCENTAGE: str = "percentage"
    FIXED: str = "fixed"


class DiscountApplyType(str, Enum):
    PRODUCT: str = "product"
    ORDER: str = "order"
    SHIPPING: str = "shipping"


class DiscountStatus(str, Enum):
    ACTIVE: str = "active"
    SCHEDULED: str = "scheduled"
    PAUSED: str = "paused"
    ENDED: str = "ended"


class DiscountTargetType(str, Enum):
    CATEGORY: str = "category"
    PRODUCT: str = "product"
